echo/echo.c

```diff
@@ -1,15 +1,164 @@
 #include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 
-void echo (){
-    return;
+/* Escapes for -e
+    \\     backslash
+    \a     alert (BEL)
+    \b     backspace
+    \c     produce no further output
+    \e     escape
+    \f     form feed
+    \n     new line
+    \r     carriage return
+    \t     horizontal tab
+    \v     vertical tab
+    \0NNN  byte with octal val
+*/
+
+enum Flags {flag_n, flag_e};
```

Comment:
this works but it seems a bit brittle.
What about using a struct to keep the flag status?

```c
typedef struct {
    bool e;
    bool n;
} Flags;

int parse_flags(..., Flags *flags) {
   ...
   flags->e = True;
   flags->n = False;
}
```